### ISSUE TITLE
[BUG] Parameter names of generated constructor are not recognized from my own annotation processor in JDK8.

### ISSUE THREAD

**Issue by itanoss** [rzwitserloot/lombok#2405]
**Describe the bug**
I want to create annotation processor collaborated with lombok. In JDK 8, constructor parameter names of target class is different from the actual. They are `arg0`, `arg1` and so on. In JDK9+, it is properly recognized.

**To Reproduce**
Create a class with lombok constructor annotation like the following.
```java
@Data
public class SomeData {
    private final int id;
    private final String name;
}
```

Create annotation processor newly. In the first round, put the class into a list named `deferredElements` in order to attempt this process after lombok processing.

In the second round, pick the class from the list and check the parameter names from the generated constructor.

**Expected behavior**
Parameter names should be recognized properly even with JDK8.

**Version info (please complete the following information):**
 - Lombok version : 1.18.12
 - Platform : Gradle 6.0.1, Azul JDK 8.0.242, 9.0.7 and 11.0.6

**Additional context**
Full example is in [my repo](https://github.com/itanoss/example-annotation-processor-with-lombok).
you can see via `./gradlew assemble`.

Under JDK8:
```
$ ./gradlew clean assemble
...
Note: Constructors: [SomeValue(int,java.lang.String)]
Note: Parameter names: [arg0, arg1]
Note: Constructors: [SomeData(int,java.lang.String)]
Note: Parameter names: [arg0, arg1]
```

Under JDK9, 10 and 11:
```
$ ./gradlew clean assemble
...
Note: Constructors: [SomeValue(int,java.lang.String)]
Note: Parameter names: [id, name]
Note: Constructors: [SomeData(int,java.lang.String)]
Note: Parameter names: [id, name]
```

I wonder if there is another approach to achieve what I want.

**Comment by Rawi01** [comment]
I think that this one was fixed in #2556, can you try it again using the latest lombok version?

**Comment by itanoss** [comment]
@Rawi01 Oh, it works now. Thank you.
